lib/codegen/src/context.rs

```diff
@@ -124,6 +125,7 @@ impl Context {
         if isa.flags().opt_level() != OptLevel::Fastest {
             self.preopt(isa)?;
         }
+        self.canonicalize_nans(isa)?; // FIXUP: Should this be wrapped in a conditional?
```

Comment:
Yes, this should be guarded by `if isa.flags().enable_nan_canonicalization`, so that we don't do the canonicalization unless we're asked to.